middlewares/osio/auth.go

```diff
@@ -120,9 +214,22 @@ func extractToken(auth string) (string, error) {
 	return auths[len(auths)-1], nil
 }
 
-func cacheKey(token string) string {
+func getService(req *http.Request) string {
+	reqPath := req.URL.Path
+	switch {
+	case strings.HasPrefix(reqPath, "/"+api):
```

Comment:
I would feel better if we determine Service based on ServiceAccount name in token instead of path+headers...